MdeModulePkg/Universal/HiiDatabaseDxe/Image.c

```diff
@@ -549,13 +549,13 @@ ImageToBlt (
     OffsetY1 = Width * Ypos;
     OffsetY2 = ImageOut->Width * (BltY + Ypos);
     for (Xpos = 0; Xpos < Width; Xpos++) {
-      SrcPixel = BltBuffer[OffsetY1 + Xpos];
+      SrcPixel.Raw = *(UINT32 *)(&BltBuffer[OffsetY1 + Xpos]);
       if (Transparent) {
         if (CompareMem (&SrcPixel, &ZeroPixel, 3) != 0) {
-          ImageOut->Image.Bitmap[OffsetY2 + BltX + Xpos] = SrcPixel;
+          WriteUnaligned32 ((UINT32 *)&ImageOut->Image.Bitmap[OffsetY2 + BltX + Xpos], SrcPixel.Raw);

```

Comment:
Why are `WriteUnaligned32()` calls required?  Can `ImageOut->Image.Bitmap` be unaligned?

Also do not use UINT32 *.  

```
EFI_GRAPHICS_OUTPUT_BLT_PIXEL_UNION *ImageBitmap;

ImageBitmap = (EFI_GRAPHICS_OUTPUT_BLT_PIXEL_UNION *)ImageOut->Image.Bitmap;

ImageBitmap[OffsetY2 + BltX + Xpos].Raw = SrcPixel.Raw;
```

Reply:
I don't think it's necessary after looking at it again. Removed it.